sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java

```diff
@@ -342,7 +375,83 @@ public String getEntityPath() {
-            .flatMap(managementNode -> managementNode.cancelScheduledMessage(sequenceNumber, linkName.get()));
+            .flatMap(managementNode -> managementNode.cancelScheduledMessage(sequenceNumber, linkName.get(),
+                null));
+    }
+
+    /**
+     * Cancels the enqueuing of an already scheduled message, if it was not already enqueued.
+     *
+     * @param sequenceNumber of the scheduled message to cancel.
+     * @param transactionContext to be set on batch message before scheduling them on Service Bus.
+     *
+     * @return The {@link Mono} that finishes this operation on service bus resource.
+     *
+     * @throws IllegalArgumentException if {@code sequenceNumber} is negative.
+     */
+    public Mono<Void> cancelScheduledMessage(long sequenceNumber, ServiceBusTransactionContext transactionContext) {
+        if (sequenceNumber < 0) {
+            return monoError(logger, new IllegalArgumentException("'sequenceNumber' cannot be negative."));
+        }
+
+        return connectionProcessor
+            .flatMap(connection -> connection.getManagementNode(entityName, entityType))
+            .flatMap(managementNode -> managementNode.cancelScheduledMessage(sequenceNumber, linkName.get(),
+                transactionContext));
+    }
+
+    /**
+     * Cancels the enqueuing of an already scheduled message, if it was not already enqueued.
+     *
+     * @param sequenceNumbers of the scheduled messages to cancel.
+     *
+     * @return The {@link Mono} that finishes this operation on service bus resource.
+     *
+     * @throws NullPointerException if {@code sequenceNumbers} is null.
+     */
+    public Mono<Void> cancelScheduledMessages(Iterable<Long> sequenceNumbers) {
+
+        if (isDisposed.get()) {
+            return monoError(logger, new IllegalStateException(
+                String.format(INVALID_OPERATION_DISPOSED_RECEIVER, "cancelScheduledMessages")));
+        }
+
+        if (Objects.isNull(sequenceNumbers)) {
+            return monoError(logger, new NullPointerException("'messages' cannot be null."));
+        }
+
+        return connectionProcessor
+            .flatMap(connection -> connection.getManagementNode(entityName, entityType))
+            .flatMap(managementNode -> managementNode.cancelScheduledMessages(sequenceNumbers, linkName.get(),
+                null));
+    }
+
+    /**
+     * Cancels the enqueuing of an already scheduled message, if it was not already enqueued.
+     *
+     * @param sequenceNumbers of the scheduled messages to cancel.
+     * @param transactionContext to be set on batch sequence numbers for this operation on Service Bus.
+     *
+     * @return The {@link Mono} that finishes this operation on service bus resource.
+     *
+     * @throws NullPointerException if {@code sequenceNumbers} is null.
+     */
+    public Mono<Void> cancelScheduledMessages(Iterable<Long> sequenceNumbers,
```

Comment:
I don't remember. Did cancelling work with a transaction?

Reply:
I tested with `cancelScheduledMessages`  with rollback and it worked.  Now when I test withy commit, it is not giving me expected result.  Since we have doubt about this API, I will remove it.